File: java/src/main/java/com/maestro/agent/PiAgent.java
package com.maestro.agent;

import java.util.Arrays;
import java.util.List;
import java.util.concurrent.CompletableFuture;

public class PiAgent {
    public CompletableFuture<List<String>> scanMonorepoWorkspaces(String directory) {
        return CompletableFuture.supplyAsync(() -> {
            System.out.println("Scanning monorepo workspaces in " + directory + "...");
            try { Thread.sleep(200); } catch (InterruptedException e) {}
            return Arrays.asList("packages/core", "packages/web", "apps/desktop");
        });
    }
}

File: src/server/agents/PiAgent.ts
export class PiAgent {
	constructor() {}

	public async scanMonorepoWorkspaces(directory: string): Promise<string[]> {
		console.log(`Scanning monorepo workspaces in ${directory}...`);
		return new Promise((resolve) => {
			setTimeout(() => {
				resolve(['packages/core', 'packages/web', 'apps/desktop']);
			}, 200);
		});
	}
}
